### ISSUE TITLE
build system picks up uapi/gsx.h  in kernel tree instead of its intended user location

### ISSUE THREAD

**Issue by aep** [gramineproject/gramine#498]
```
Has header "/usr/src/linux-headers-5.16.11-arch1-2/arch/x86/include/uapi/asm/sgx.h" : NO
``

i don't think that's correct. /usr/src is for kernel modules, not for userspace application.
uapi is in /usr/include/asm

specifically trying to pick up the kernel source for the _running_ kernel suggests there's some fundamental misunderstanding of how stable uapi is supposed to be

**Comment by boryspoplawski** [comment]
https://github.com/gramineproject/gramine/blob/master/meson.build#L91 `/usr/include` is preferred.

**Comment by aep** [comment]
thanks. this makes sense actually. unclear why this is broken for me then. i'll try to strace meson.

**Comment by boryspoplawski** [comment]
How did you configure meson (what was the cmdline)?

**Comment by boryspoplawski** [comment]
Yeah, non-ubuntu hosts are poorly tested and are not officially supported.
Note that before SGX driver was upstreamed (before v5.11) there was nothing stable about those headers and we had to actually build against sources of whatever is running at the moment.

I asked for meson cmdline, because we have `sgx_driver_include_path` option, which allows you to point to whatever directory you want - this should be suitable for you.

**Comment by aep** [comment]
> sgx_driver_include_path

oh, of course. can confirm this works for me, and it's probably the correct fix.
thanks!
